File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    let stack = [];
    let answer = '';
    
    for (let i=0; i<s.length; i++) {
        if (s[i] == '(') {
            stack.push(s[i]);
            if (stack.length > 1) {
                answer += s[i];
            }
        } else {
            if (stack.length > 1) {
                answer += s[i];
            }
            stack.pop();
        }

    }

    return answer;
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.py
class Solution(object):
    def removeOuterParentheses(self, s):
        """
        :type s: str
        :rtype: str
        """
        answer = ''
        level = 0
        for i in s:
            if i == "(":
                level += 1
                if level > 1:
                    answer += i
            if i == ")":
                if level > 1:
                    answer += i
                level -= 1

        return answer     